api/api_controller.go

```diff
@@ -1301,42 +1302,64 @@ func (c *Controller) ObjectsDeleteObjectHandler() objects.DeleteObjectHandler {
 	})
 }
 
-func (c *Controller) RevertBranchHandler() branches.RevertBranchHandler {
-	return branches.RevertBranchHandlerFunc(func(params branches.RevertBranchParams, user *models.User) middleware.Responder {
+func (c *Controller) RevertHandler() branches.RevertHandler {
+	return branches.RevertHandlerFunc(func(params branches.RevertParams, user *models.User) middleware.Responder {
 		deps, err := c.setupRequest(user, params.HTTPRequest, []permissions.Permission{
 			{
 				Action:   permissions.RevertBranchAction,
 				Resource: permissions.BranchArn(params.Repository, params.Branch),
 			},
 		})
 		if err != nil {
-			return branches.NewRevertBranchUnauthorized().WithPayload(responseErrorFrom(err))
+			return branches.NewRevertUnauthorized().WithPayload(responseErrorFrom(err))
 		}
 		deps.LogAction("revert_branch")
-		cataloger := deps.Cataloger
+		err = deps.Cataloger.Revert(deps.ctx, params.Repository, params.Branch, params.Revert.Ref)
+		if errors.Is(err, db.ErrNotFound) {
+			return branches.NewRevertNotFound().WithPayload(responseErrorFrom(err))
+		}
+		if err != nil {
+			return branches.NewRevertDefault(http.StatusInternalServerError).WithPayload(responseErrorFrom(err))
+		}
+		return branches.NewRevertNoContent()
+	})
+}
 
+func (c *Controller) ResetBranchHandler() branches.ResetBranchHandler {
+	return branches.ResetBranchHandlerFunc(func(params branches.ResetBranchParams, user *models.User) middleware.Responder {
+		deps, err := c.setupRequest(user, params.HTTPRequest, []permissions.Permission{
+			{
+				Action:   permissions.RevertBranchAction,
```

Comment:
You're using the Revert permissions here on purpose?

Reply:
Yes, didn't want to add an extra set of permissions